crates/cli-support/src/js/binding.rs

```diff
@@ -987,18 +1001,32 @@ fn instruction(js: &mut JsBuilder, instr: &Instruction, log_error: &mut bool) ->
         }
 
         Instruction::RustFromI32 { class } => {
-            js.cx.require_class_wrap(class);
             let val = js.pop();
-            js.push(format!("{}.__wrap({})", class, val));
+            if let Some(name) = constructor {
+                if name != class {
+                    bail!("constructor for `{}` cannot return `{}`", name, class);
+                }
+                js.push(format!("{} >>> 0;", val));
+            } else {
+                js.cx.require_class_wrap(class);
+                js.push(format!("{}.__wrap({})", class, val));
+            }
         }
 
         Instruction::OptionRustFromI32 { class } => {
-            js.cx.require_class_wrap(class);
             let val = js.pop();
-            js.push(format!(
-                "{0} === 0 ? undefined : {1}.__wrap({0})",
-                val, class,
-            ))
+            if let Some(name) = constructor {
+                if name != class {
+                    bail!("constructor for `{}` cannot return `{}`", name, class);
+                }
+                js.push(format!("{val} === 0 ? 0 : ({val} >>> 0);"));
```

Comment:
I mentioned in the broader review comment that I don't think `Option<T>` should be allowed anyway, but I'm also not too sure what this is trying to accomplish. Setting `__wbg_ptr` to 0 will simply cause any method calls on the type to fail, since that's what we set it too to mark that the struct's been freed, which doesn't seem super useful to me; there's no good way to check if it's `None`.

Reply:
Yeah, I wasn't sure what to do this about this either. In the current implementation sets the instance's `__wbg_ptr` is `undefined` since it never calls `__wrap`. I'm not sure if this is better tbh!